### PR TITLE
Enable store_failures for tests

### PR DESCRIPTION
**Pull request by 0xRobin**
This enables `store_failures` which allows users to directly query the failing test results from a PR.

- I had to truncate the custom aliases as there is a max of 128 chars for table names.
- column names from tests have to be unique and can't contain special chars: `' ,;{}()\n\t='`

Example run: https://github.com/duneanalytics/spellbook/actions/runs/3960021058/jobs/6783593246
~~The CI on this PR might time out since all the tests are considered modified with this config change~~

### CODE REVIEW

**Comment by 0xRobin** [comment]
rerunning the tests with longer timeout as it was really close to finishing in 1h and there are some errors we should look into.

**Comment by 0xRobin** [comment]
@dot2dotseurat @jeff-dude @soispoke 
Managed to get this one working, [the latest CI run](https://github.com/duneanalytics/spellbook/actions/runs/3966551950/jobs/6797487900) shows the expected results. :partying_face: 

Still need to revert the CI timeout back to 1h after your reviews.

**Comment by soispoke** [comment]
looks good to me, but it'll probably be good to have @dot2dotseurat's review as well for this one!

**Review by soispoke — APPROVED** [review]

**Comment by 0xRobin** [comment]
@dot2dotseurat bump, any chance we can move this forward? :pray:

**Review by dot2dotseurat — APPROVED** [review]
Assuming we re-lower the time out to 60, looks good. 

This is very helpful! Great work.

File: .github/workflows/dbt_slim_ci.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ concurrency:
 jobs:
   dbt-test:
     runs-on: [ self-hosted, linux, spellbook ]
-    timeout-minutes: 60
+    timeout-minutes: 120
```
[the remark this answers is not in the document]
Comment (0xRobin) [reply .github/workflows/dbt_slim_ci.yml:23]: it's not, storing the test results does not significantly increase the CI runtime.

**Pull request merged**